### ISSUE TITLE
Edictor does not save the data after editing alignments

### ISSUE THREAD

**Issue by thiagochacon** [digling/edictor#62]
In the Huber data, many times I tried to save the data after alignments are edited (saving is automatically) an error pops up saying it was impossible to save the data. Analyses have been lost in some points

**Comment by LinguList** [comment]
Which browser are you using? With firefox it's supposed to work, with
chrome and safari, and the windows thing, there may be problems. make
sure to always clean your cache in firefox or open the edictor in
private mode surfing, so that the settings are cleaned.

Please let me know to which degree this helps. I'll look into the
problem ASAP, once I know more reg. the questions stated above.

**Comment by LinguList** [comment]
Update here; we have a definite problem in storing '#' character. This should either be deleted from all files we have, or we need to make sure it is escaped. Even easier on the long run might be switching to post statements instead of get statements...

**Comment by LinguList** [comment]
Just for convenience, I paste the link where it failed [here](http://tsv.lingpy.org/?file=huber1992&remote_dbase=huber1992&columns=DOCULECT|CONCEPT|COUNTERPART|TOKENS|ALIGNMENT|COGID&basics=DOCULECT|CONCEPT|COUNTERPART|TOKENS|ALIGNMENT|COGID&formatter=COGID&doculects=bora|mira%C3%B1a|muinane|ocaina|carijona|puinave|achagua|yucuna|curripaco|jupda|kakua|koreguaje|orej%C3%B3n|wanano|tucano|tanimuca|piapoco|barasana|siona|sekoya|nukak|cubeo|baniva)

**Comment by LinguList** [comment]
And for the record: the problems happen in cognate set mode, which may be a hint regarding the problems or not. First problems occured between "caliente" and "arder burn" (btw. the order of the concepts in cognate set mode definitely needs to be fixed, ASAP, but maybe best along with this issue).

**Comment by LinguList** [comment]
summarizing the last issues here (mentioned by @thiagochacon ):

- some languages have words in spanish, which are clear not borrowings but information from footnotes
- indeed, sometimes footnotes have been added as "words" in an inconsistent way (I could delete those manually in EDICTOR)
- when working in cognate mode, if I choose a word, work ont it and than use the --> arrow for the next word it will not go to the next word exactly, but will go back to the first word in the cognate mode list of concepts

**Comment by LinguList** [comment]
I think this could be solved now, that I worked on the cognate set mode for quite some time and also updated the way the data is stored (everything is safely escaped right now). I won't close this for the moment, however, since I think it's better to wait whether new problems arise here.

**Comment by LinguList** [comment]
Just cleaned the data, as discussed, with a slight modification that makes it easier for me:

* I split off all values with a "_" underscore, but I didn't add new word entries for them.
* the idea is: for huber1992, one can instead look for those cases which have '!' in the note-field (just sort the notefield by double-clicking, for example), and then look, whether it's OK to only retain the first part, or whether the second part is also needed. This can be best done by looking either into the "counterpart" or the "ipa" field, which retain this original entry with both words.

There are quite a few such cases: 2500.

for the selection, which I now also updated, there are some 700 such cases.

Here's the link including  the notes field for the selection discussed: [the link](http://tsv.lingpy.org/?file=huber1992&remote_dbase=huber1992&columns=DOCULECT|CONCEPT|COUNTERPART|TOKENS|ALIGNMENT|COGID|NOTE&basics=DOCULECT|CONCEPT|COUNTERPART|TOKENS|ALIGNMENT|COGID|NOTE&formatter=COGID&doculects=bora|mira%C3%B1a|muinane|ocaina|carijona|puinave|achagua|yucuna|curripaco|jupda|kakua|koreguaje|orej%C3%B3n|wanano|tucano|tanimuca|piapoco|barasana|siona|sekoya|nukak|cubeo|baniva)

Please make sure that the data was correctly inserted, @thiagochacon . If not, letme know, I made a backup of everything. If yes, and if storing works again, please close this issue.

**Comment by LinguList** [comment]
BTW: If the link above doesn't work the first time, click the refresh button in the browser, or go with the cursor in the URL line and hit enter. This should work by then.

**Comment by thiagochacon** [comment]
When doing alignments (in the alignment pop up window) the data is submitted and EDICTOR says the data was saved, however no apparent change has happened in "alignment" itslef (i.e. in the alignment column, not in the alignment pop up window). It is only after refreshing the browser window that the changes appear in the actual alignment column.
Problem with data visualization only?

**Comment by LinguList** [comment]
Yes, only visualization: I was careful this time to not refresh immediately, since in cases where one selects multiple concepts (for cross-semantic cognates), I need to expand the code explicitly to remember that one was in multi-concept mode. I was rushing in getting it online to see whether changes are now regularly stored, but I should go and have it also immediately refreshed, which is not that problematic. In the meantime, it is enough to press on the button between the arrows in cognate set mode (the one showing to left and right) to get the cognate set view refreshed. 

Could you make another issue for this, so that I don't forget to fix it? I'll try and have it done during this week.

**Comment by LinguList** [comment]
Allright, I close this for the time being, since I have the imporession, the problem is solved. Please reopen, in case it re-appears...

**Comment by LinguList** [comment]
Ok, we cannot sufficiently avoid if a url is too long that this will be handled without shortening urls.
However, by switching to post-requests, when communicating witht he python server, this is generally saver. So with the next commit, this issue will be solved.
